scrunch/helpers.py

```diff
@@ -70,7 +76,50 @@ def download_file(url, filename):
     return filename
 
 
-def case_expr(rules, name, alias):
+def validate_categories(categories):
+    """
+    Categories can be defined as simple as:
+        categories=[
+            {'id': 1, 'name': 'Yes', 'selected': True},
+            {'id': 2, 'name': 'No'},
+            {'id': 3, 'name': 'Maybe'},
+            {'id': 4, 'name': 'Missing', 'missing': True}
+        ]
+    This method takes care of validating that only 1 is selected, that
+    only one or None is missing and to fill the definition above to
+    match the API requirement:
+        categories=[
+            {'id': 1, 'name': 'Yes', 'missing': False, 'numeric_value': None, 'selected': True},
+            {'id': 2, 'name': 'No', 'missing': False, 'numeric_value': None, 'selected': False},
+            {'id': 3, 'name': 'Maybe', 'missing': False, 'numeric_value': None, 'selected': False},
+            {'id': 4, 'name': 'Missing', 'missing': True, 'numeric_value': None, 'selected': False}
+        ]
+    """
+    defaults = {'missing': False, 'numeric_value': None, 'selected': False}
+    selected_count = 0
+    missing_count = 0
+    for category in categories:
+        if category.get('selected'):
+            selected_count += 1
+        if category.get('missing'):
+            missing_count += 1
+        if not category.get('id'):
+            raise ValueError('An "id" must be provided to all categories')
+        if not category.get('name'):
+            raise ValueError('A "name" must be provided to all categories')
+    if selected_count > 1 or selected_count == 0:
+        raise ValueError('Categories must define one category as selected')
+    if missing_count > 1:
+        raise ValueError('Categories can\'t have more than 1 category as missing')
```

Comment:
Is this so? There is no restriction on the backend for this rule. You can have all the missing categories you want.